multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/StagingApplicationAttributeUpdater.java

```diff
@@ -3,46 +3,90 @@
     }
 
     @Override
-    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplication application) {
+    @SuppressWarnings("unchecked")
+    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplicationExtended application) {
         Staging staging = application.getStaging();
-        Staging existingStaging = existingApplication.getStaging();
-        return hasStagingChanged(staging, existingStaging);
+        boolean isSshEnabled = getControllerClient().getApplicationSshEnabled(existingApplication.getGuid());
+        DockerInfo dockerInfo = null;
+        List<String> buildpacks = null;
+        String stack = null;
+        var lifecycle = existingApplication.getLifecycle();
+        if (lifecycle.getType() == LifecycleType.BUILDPACK) {
+            buildpacks = (List<String>) lifecycle.getData()
+                                                 .get("buildpacks");
+            stack = (String) lifecycle.getData()
+                                      .get("stack");
+        } else {
+            var droplet = getControllerClient().getCurrentDropletForApplication(existingApplication.getGuid());
+            var cloudPackage = getControllerClient().getPackage(droplet.getPackageGuid());
+            var dockerData = (DockerData) cloudPackage.getData();
+            dockerInfo = ImmutableDockerInfo.builder()
+                                            .image(dockerData.getImage())
+                                            .build();
+            if (dockerData.getUsername() != null && dockerData.getPassword() != null) {
+                dockerInfo = ImmutableDockerInfo.copyOf(dockerInfo)
+                                                .withCredentials(ImmutableDockerCredentials.builder()
+                                                                                           .username(dockerData.getUsername())
+                                                                                           .password(dockerData.getPassword())
+                                                                                           .build());
+            }
+        }
+        return hasStagingChanged(staging, buildpacks, stack, dockerInfo, isSshEnabled);
     }
 
-    private boolean hasStagingChanged(Staging staging, Staging existingStaging) {
+    private boolean hasStagingChanged(Staging staging, List<String> existingBuildpacks, String existngStack, DockerInfo existingDockerInfo,
+                                      boolean existingSshEnabled) {
         List<String> buildpacks = staging.getBuildpacks();
         String command = staging.getCommand();
         String stackName = staging.getStackName();
+        String healthCheckType = staging.getHealthCheckType();
+        if (healthCheckType == null) {
+            healthCheckType = "port"; //use default health check type https://v3-apidocs.cloudfoundry.org/version/3.119.0/#the-health-check-object
```

Comment:
How it is handled now? I think we set to health check type to `port` now but maybe in different step.